src/transport/http/client.rs

```diff
@@ -73,7 +73,16 @@ impl HttpTransportClient {
         // Because hyper can only be polled through tokio, we spawn it in a background thread.
         thread::Builder::new()
             .name("jsonrpsee-hyper-client".to_string())
-            .spawn(move || background_thread(requests_rx))
+            .spawn(move || {
+                let client = hyper::Client::new();
+                background_thread(requests_rx, move |rq| {
+                    // cloning Hyper client = cloning references
+                    let client = client.clone();
+                    async move {
+                        let _ = rq.send_back.send(client.request(rq.request).await);
+                    }
+                })
+            })
```

Comment:
The polling fix looks good, but I don't understand this change? Why not leave `hyper::Client::new()` in the background thread?

Reply:
To test that `background_thread` is actually doing something - see `background_thread_is_able_to_complete_requests`.